lib/AST/GenericEnvironment.cpp

```diff
@@ -124,7 +124,7 @@ Type GenericEnvironment::mapTypeIntoContext(GenericEnvironment *env,
 Type MapTypeOutOfContext::operator()(SubstitutableType *type) const {
   auto archetype = cast<ArchetypeType>(type);
   if (isa<OpaqueTypeArchetypeType>(archetype->getRoot()))
-    return type;
+    return Type();
```

Comment:
My understanding is that return null type here indicates that archetype can't be mapped out from opaque type instead of just preserving the original archetype which could lead to some other changes in the underlying opaque type? Is that correct?

Reply:
AFAICT, it uses the substitution function returning null to indicate that it needs to fall back to structural replacement:

https://github.com/apple/swift/blob/master/lib/AST/Type.cpp#L3281

Reply:
Thanks!